scripts/analyze-ios-network-log.py

```
@@ -0,0 +1,262 @@
+            liveness_count += 1
+
+    duplicate_peers = {
+        peer: count for peer, count in max_active.items() if count > 1
+    }
+    failures: list[str] = []
+    if not connected_latencies and not usable_latencies:
+        failures.append("no successful transport or usable RPC connection")
+    if duplicate_peers:
+        failures.append(f"more than one active physical session for {sorted(duplicate_peers)}")
+    if expected_background_seconds > 0 and background_gaps:
+        if max(background_gaps) < expected_background_seconds:
```

Ruff (0.16.1)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)